src/operations/scripts.rs

```diff
@@ -0,0 +1,200 @@
+    ];
+
+    for name in names.iter() {
+        process_script_with_name(venv_path, scripts_path, &name, mode)?;
+    }
+    Ok(())
+}
+
+fn process_script_with_name(
+    venv_path: &PathBuf,
+    scripts_path: &PathBuf,
+    name: &str,
+    mode: &ProcessScriptsMode,
+) -> Result<(), Error> {
+    let src_path = venv_path.join(SCRIPTS_SUBDIR).join(name);
+    let dest_path = scripts_path.join(name);
+    cmd::print_info_2(&format!("Creating script: {}", name.bold()));
+    if !src_path.exists() {
+        return Err(new_error(&format!(
+            "{:?} does not exist. You may want to call `dmenv develop` now",
+            src_path
+        )));
+    }
+    #[cfg(windows)]
+    {
+        match mode {
+            Safe => safe_copy(&src_path, &dest_path),
+            Override => copy(&src_path, &dest_path),
+        }
+    }
+    #[cfg(unix)]
+    {
+        symlink(&src_path, &dest_path, mode)
+    }
+}
+
+#[cfg(windows)]
+fn safe_copy(src_path: &PathBuf, dest_path: &PathBuf) -> Result<(), Error> {
+    if dest_path.exists() {
+        return Err(new_error(&format!("{:?} already exists", src_path)));
+    }
+    copy(src_path, dest_path)
+}
+
+#[cfg(windows)]
+fn copy(src_path: &PathBuf, dest_path: &PathBuf) -> Result<(), Error> {
+    std::fs::copy(src_path, dest_path).map_err(|e| {
+        new_error(&format!(
+            "Could not copy from {:?} to {:?}: {}",
+            src_path, dest_path, e
+        ))
+    })?;
+    Ok(())
+}
+
+#[cfg(unix)]
+fn symlink(
+    src_path: &PathBuf,
+    dest_path: &PathBuf,
+    mode: &ProcessScriptsMode,
+) -> Result<(), Error> {
+    match mode {
+        // Note: we assume it is "safe" to change the target of an existing
+        // symlink
+        Safe => delete_if_link(dest_path),
+        // Note: we need to delete dest_path for unix::fs::symlink to work
+        // later on
+        Override => delete_if_exists(dest_path),
+    }?;
+    println!(
+        "{} -> {}",
+        dest_path.to_string_lossy(),
+        src_path.to_string_lossy()
+    );
+    std::os::unix::fs::symlink(src_path, dest_path).map_err(|e| {
+        new_error(&format!(
+            "Could not create link from {:?} to {:?}: {}",
+            dest_path, src_path, e
+        ))
+    })
+}
+
+#[cfg(unix)]
+fn delete_if_link(path: &PathBuf) -> Result<(), Error> {
+    // This will make an error if the path does not exist,
+    // but also if we don't have permission to read the path.
+    // In both cases, we want to return early - in the later,
+    // the code will probably fail when creating the symlink
+    // later on.
+    let meta = std::fs::symlink_metadata(path);
+    if meta.is_err() {
+        return Ok(());
+    };
+    let meta = meta.unwrap();
+    if !meta.file_type().is_symlink() {
+        return Err(new_error(&format!(
+            "{:?} exists and is *not* a symlink",
+            path
+        )));
+    };
+    std::fs::remove_file(path).map_err(|e| {
```

Comment:
Note that in theory the functions around here are all kinds of unsafe, you have some pretty glaring TOCTOU problems in this file =]
(That said, I can understand that it's probably not a very threatening concern in practice in the case of dmenv: there's no security implications and probably nobody's going to be touching those files while dmenv runs anyways. I'd just be careful not to reuse this potentially unsafe function everywhere without thinking about it. Filesystems are hard!)

Reply:
I agree, but those functions are private :)

Reply:
Sure!